scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -1574,3 +1574,26 @@ def func(x):
         # after the '=', so if necessary, the text could be reduced to, say,
         # "MAXCV = 0.".
         assert "MAXCV = 0.414" in result.message
+
+    def test_strategy_func(self):
+
+        init = [(-8, 0.5)] * 5
+        bounds = [(-8.0,8.0)] * 2
+
+        def func(x):
+            return abs(x[0]) * abs(x[1])
+            
+        def custom_strategy_fn(candidate, population, rng):
+            assert(np.array_equal(candidate, [-8, 0.5]) or np.array_equal(candidate, [2.0, -1.0]))
```

Comment:
- candidate is an integer.
- `assert` is not a function.

```suggestion
            assert 0 < candidate < len(population)
```

In general it's not a good idea to use `np.testing.assert_equal` or `np.array_equal` because of floating point arithmetic. Use `np.testing.assert_allclose` instead.

Reply:
Should I commit this change even though it takes out the test of whether the arrays match either the (A) initial or (B) new candidate, rather than changing them to assert_allclose?

You'll note that the floating point numbers I chose in the test are all powers of two. Powers of two should always match exactly.

Reply:
_"assert is not a function"_ - LOL. I've been using it for years like that and never realized that it's just treating the parentheses as grouping parentheses  ;)  Great to know!